In [356]:
import pandas as pd
import matplotlib as plt

## Set

In [357]:
def carregar_arquivo_excel(caminho, sheet_name=0, skiprows=None):
    return pd.read_excel(caminho, sheet_name=sheet_name, skiprows=skiprows)

In [358]:
caminhos = {
'fup' : r'C:\Users\fsp_adolpho.salvador\Desktop\Konica Minolta\Desktop Cloud - Documentos\Desktop\Py\PowerBI\Novo\FUP.XLSX',
'zstok':  r'C:\Users\fsp_adolpho.salvador\Desktop\Konica Minolta\Desktop Cloud - Documentos\Desktop\Py\PowerBI\Novo\ZSTOK.XLSX',
'zmb51' : r'C:\Users\fsp_adolpho.salvador\Desktop\Konica Minolta\Desktop Cloud - Documentos\Desktop\Py\PowerBI\Novo\saidas geral.xlsx',
}

In [359]:
fup = carregar_arquivo_excel(caminhos['fup'])
zstok = carregar_arquivo_excel(caminhos['zstok'])
zmb51 = carregar_arquivo_excel(caminhos['zmb51'])

In [360]:
zstok.columns	

Index(['Tipo de material', 'Material', 'Descrição', 'UMB', 'Cen.', 'Dep.',
       'Pos.dpst.', 'Utiliz.livre', 'Contr.qualid.', 'Transf.dep.',
       'Transfer.(centro)', 'Bloqueado', 'Estoque Total', 'Reservado',
       'Remessa->Cliente', 'SC', 'Status todos centros', 'SV',
       'Status vendas'],
      dtype='object')

In [361]:
zstok_mat = zstok[['Material','Tipo de material']]
zstok_mat.columns	

Index(['Material', 'Tipo de material'], dtype='object')

In [362]:
zstok_mat.set_index('Material')

Tipo de material
Material                             
3405        Produto Coml-Acess.- Corr
4030        Produto Coml-Acess.- Corr
4363        Produto Coml-Acess.- Corr
10307       Produto Coml-Acess.- Corr
4061112     Produto Coml-Acess.- Corr
...                               ...
ZA104501A0  Produto Coml-Peça-SPO/MAO
ZA10458100  Produto Coml-Peça-SPO/MAO
ZA10458100  Produto Coml-Peça-SPO/MAO
ZA10461900  Produto Coml-Peça-SPO/MAO
NaN                               NaN

[14577 rows x 1 columns]

In [363]:
fup.columns

Index(['Nº Sequência do Embarque', 'Nº do Pedido', 'Data Pedido', 'Centro',
       'Depósito', 'Pos.Dep.MM', 'Pos.Dep.WM', 'Cod. Fornecedor',
       'Nome Fornecedor', 'Item', 'Saldo Pedido', 'UMP', 'UMP.1', 'UMP.2',
       'Material', 'Texto Breve', 'Qtde Pedido', 'Vl. Total Pedido', 'Moeda',
       'Data de Remessa', 'Data Estatística', 'Sua Referência',
       'Nossa Referência', 'Embarque', 'Meio de Transporte', 'Meio Embarque',
       'Nº do Conhecimento', 'Data do BL', 'Data de Chegada', 'Declaração',
       'Dt.Registro', 'Dt.Desembaraço', 'Qtde. Desembarcada',
       'Referência de Fatura', 'Data da Fatura', 'Valor da Fatura',
       'Data Prev.Entrada', 'Dt. Recebimento', 'Recebimento', 'Peso Líquido',
       'Un', 'Cód.NCM', 'Fabricante', 'Num.conversão', 'Qtd.Ped.UMB',
       'TipoMat'],
      dtype='object')

In [364]:
fup_filtro = fup[['Material','Qtde Pedido','Nº do Pedido','Data Prev.Entrada','TipoMat','Embarque']]

In [365]:

fup_filtro = fup_filtro[fup_filtro['Embarque'] == 'IMP 006-25']

In [366]:
fup_filtro

Material  Qtde Pedido  Nº do Pedido Data Prev.Entrada TipoMat    Embarque
304   A7U40KD            4    5000003853        2025-04-11    PCCC  IMP 006-25
305   A0X5235            4    5000003853        2025-04-11    PCCC  IMP 006-25
306   A0X5335            1    5000003853        2025-04-11    PCCC  IMP 006-25
307   AEW6130            8    5000003853        2025-04-11    PCCC  IMP 006-25
308   AEW6230            8    5000003853        2025-04-11    PCCC  IMP 006-25
...       ...          ...           ...               ...     ...         ...
1150  AE2F430           29    5000003892        2025-04-11    PCCC  IMP 006-25
1151  AE2G700            4    5000003892        2025-04-11    PCCC  IMP 006-25
1152  AE2G800            2    5000003892        2025-04-11    PCCC  IMP 006-25
1153  AE2G900            5    5000003892        2025-04-11    PCCC  IMP 006-25
1154  AE7XWY1           18    5000003892        2025-04-11    PCCC  IMP 006-25

[129 rows x 6 columns]

In [367]:
fup_filtro['Embarque'].unique()

array(['IMP 006-25'], dtype=object)

In [368]:
fup_filtro['TipoMat'].unique()

array(['PCCC'], dtype=object)

In [369]:
fup_filtro['Nº do Pedido'].unique()

array([5000003853, 5000003892], dtype=int64)

In [370]:
fup_Material = fup_filtro['Material'].unique()

In [371]:
fup_filtro = pd.pivot_table(fup_filtro,
    values='Qtde Pedido',
    index='Material',
    columns='Data Prev.Entrada',
    aggfunc='sum',
    fill_value=0
)
fup_filtro

Data Prev.Entrada  2025-04-11
Material                     
14YK                       23
A0TH500                    10
A0X5134                    95
A0X5135                    11
A0X5234                    31
...                       ...
AEW6130                     8
AEW6230                     8
AEW6330                     8
AEW6430                     8
AF1R600                    28

[126 rows x 1 columns]

In [372]:
zmb51

Material                     Texto breve material Centro  \
0                02UK  CX C/2 SACOS DE REVEL EM PO BH PRO 1050   CE05   
1                1854                    MASTER CZ B4   S-4876   CE09   
2                4090        ESPELHO TAMPA BR PARA TOMADA NEMA   CE09   
3                4090        ESPELHO TAMPA BR PARA TOMADA NEMA   CE09   
4               53411     CART CYAN SURECOLOR SERIE T(I.T69420   CE03   
...               ...                                      ...    ...   
386409  T748XXL120-AL                                      NaN   CE07   
386410  T748XXL220-AL                                      NaN   CE07   
386411  T748XXL320-AL                                      NaN   CE07   
386412  T748XXL420-AL                                      NaN   CE07   
386413      TN3662XLS                                      NaN   CE07   

       Depósito  Tipo de movimento  Estoque especial  Doc.material  \
0          DEP2                601               NaN      49014982   
1          DEP2                601               NaN      49012562   
2          DEP3                601               NaN      49015466   
3          DEP3                601               NaN      49009446   
4          DEP2                952               NaN      49006627   
...         ...                ...               ...           ...   
386409     DEP7                952               NaN      49007375   
386410     DEP7                952               NaN      49007375   
386411     DEP7                952               NaN      49007375   
386412     DEP7                952               NaN      49007375   
386413     DEP7                952               NaN      49007376   

        Item doc.material Data de lançamento  Qtd.  UM registro UM registro  \
0                       1         01/07/2021                1.0         UND   
1                       2         01/06/2021                1.0         UND   
2                       2         01/07/2021                1.0         UND   
3                       2         01/05/2021                1.0         UND   
4                       1         01/03/2021                1.0         UND   
...                   ...                ...                ...         ...   
386409                  1         01/03/2025                2.0         UND   
386410                  2         01/03/2025                1.0         UND   
386411                  3         01/03/2025                2.0         UND   
386412                  4         01/03/2025                2.0         UND   
386413                  1         01/03/2025                4.0         UND   

        Cliente  Fornecedor                        Cliente Nome  \
0           307         NaN      COMAVI COMERCIO DE MAQ E VISUA   
1          8143         NaN      IGREJA PENTEC ASSEMBL. DEUS M.   
2         10791         NaN       FOX CURITIBA FOTOGRAFIAS LTDA   
3          5763         NaN      IMPRESUL SERVICO GRAF E EDIT L   
4          9561         NaN      BOEHRINGER ING ANI HEALTH DO B   
...         ...         ...                                 ...   
386409     3469         NaN  CHIBATAO NAVEGACAO E COMERCIO LTDA   
386410     3469         NaN  CHIBATAO NAVEGACAO E COMERCIO LTDA   
386411     3469         NaN  CHIBATAO NAVEGACAO E COMERCIO LTDA   
386412     3469         NaN  CHIBATAO NAVEGACAO E COMERCIO LTDA   
386413    11599         NaN      J F DE OLIVEIRA NAVEGACAO LTDA   

       Conceito de Pesquisa CONCEITO DE PESQUISA_2  \
0                    COMAVI            Indireto SP   
1                CLIENTE RS    Direto - Cliente RS   
2                CLIENTE RS    Direto - Cliente RS   
3                CLIENTE RS    Direto - Cliente RS   
4                CLIENTE SP    Direto - Cliente SP   
...                     ...                    ...   
386409           CLIENTEMAO   Direto - Cliente MAO   
386410           CLIENTEMAO   Direto - Cliente MAO   
386411           CLIENTEMAO   Direto - Cliente MAO   
3

In [373]:
zmb51['Data de lançamento'] = pd.to_datetime(zmb51['Data de lançamento'], errors='coerce')
zmb51['AnoMes'] = zmb51['Data de lançamento'].dt.to_period('M').astype(str)


In [374]:
zmb51_filtro = pd.pivot_table(zmb51, values='Qtd.  UM registro', index='Material', columns='Data de lançamento', aggfunc='sum', fill_value=0)

In [375]:
zmb51_filtro

Data de lançamento  2021-01-01  2021-01-02  2021-01-03  2021-01-04  \
Material                                                             
11                         0.0         0.0         0.0         0.0   
697                        0.0         0.0         0.0         0.0   
699                        0.0         0.0         0.0         0.0   
714                        0.0         0.0         0.0         0.0   
718                        0.0         0.0         0.0         0.0   
...                        ...         ...         ...         ...   
ZA10437580                 2.0         0.0         0.0         0.0   
ZA10438370                 0.0         0.0         0.0         0.0   
ZA1043837X                 1.0         0.0         0.0         0.0   
ZA104501A0                 0.0         0.0         1.0         1.0   
ZA10458100                 4.0         0.0         1.0         0.0   

Data de lançamento  2021-01-05  2021-01-06  2021-01-07  2021-01-08  \
Material                                                             
11                         0.0         0.0         0.0         0.0   
697                        0.0         0.0         0.0         0.0   
699                        0.0         0.0         0.0         0.0   
714                        0.0         0.0         0.0         0.0   
718                        0.0         0.0         0.0         0.0   
...                        ...         ...         ...         ...   
ZA10437580                 0.0         0.0         0.0         0.0   
ZA10438370                 0.0         0.0         0.0         0.0   
ZA1043837X                 0.0         0.0         0.0         0.0   
ZA104501A0                 0.0         0.0         0.0         1.0   
ZA10458100                 0.0         0.0         0.0         0.0   

Data de lançamento  2021-01-09  2021-01-10  ...  2024-01-07  2024-01-08  \
Material                                    ...                           
11                         0.0         0.0  ...         2.0         0.0   
697                        0.0         0.0  ...         0.0         0.0   
699                        0.0         0.0  ...         0.0         0.0   
714                        0.0         0.0  ...         0.0         0.0   
718                        0.0         0.0  ...         0.0         0.0   
...                        ...         ...  ...         ...         ...   
ZA10437580                 0.0         0.0  ...         0.0         0.0   
ZA10438370                 0.0         0.0  ...         0.0         0.0   
ZA1043837X                 0.0         0.0  ...         0.0         0.0   
ZA104501A0                 1.0         0.0  ...         1.0         0.0   
ZA10458100                 0.0         0.0  ...         0.0         0.0   

Data de lançamento  2024-01-09  2024-01-10  2024-01-11  2024-01-12  \
Material                                                             
11                         0.0         0.0         0.0         0.0   
697                        0.0         0.0         0.0         0.0   
699                        0.0         0.0         0.0         0.0   
714                        0.0         0.0         1.0         0.0   
718                        0.0         0.0         0.0         0.0   
...                        ...         ...         ...         ...   
ZA10437580                 0.0         0.0         0.0         0.0   
ZA10438370                 0.0         0.0         0.0         0.0   
ZA1043837X                 0.0         0.0         0.0         0.0   
ZA104501A0                 0.0         1.0         0.0         0.0   
ZA10458100                 0.0         0.0         0.0         0.0   

Data de lançamento  2025-01-01  2025-01-02  2025-01-03  2025-01-04  
Material                                                            
11                         0.0         0.0         0.0         0.0  
697                        0.0         0.0         0.0         0.0  
699            

In [376]:
fup_Material

array(['A7U40KD', 'A0X5235', 'A0X5335', 'AEW6130', 'AEW6230', 'AEW6330',
       'AEW6430', 'ACVT0Y2', 'A0X5435', '14YK', 'A0TH500', 'A0X5134',
       'A0X5135', 'A0X5234', 'A0X5334', 'A0X5434', 'A2X203D', 'A2X20RD',
       'A2XN03D', 'A2XN08D', 'A2XN0ED', 'A2XN0KD', 'A33K130', 'A33K232',
       'A33K330', 'A33K332', 'A33K430', 'A3VU030', 'A3VVP00', 'A3VX130',
       'A3VX230', 'A3VX231', 'A3VX330', 'A3VX331', 'A3VX430', 'A3VX431',
       'A3VX432', 'AF1R600', 'A3VX700', 'A3VX800', 'A3VX900', 'A4RCWY1',
       'A5E7130', 'A5E7230', 'A5E7330', 'A5E7430', 'A5E7600', 'A5E7700',
       'A5E7800', 'A5E7900', 'A5WH0Y0', 'A5WJ0Y0', 'A63W01F', 'A63X03W',
       'A6DY0Y1', 'A73303F', 'A73308F', 'A7330EF', 'A7330KF', 'A7U403D',
       'A7U40ED', 'A85Y03D', 'A85Y08D', 'A85Y0ED', 'A85Y0KD', 'A85Y0RD',
       'A87M030', 'A8DA130', 'A8DA230', 'A8DA330', 'A8DA430', 'A8J3130',
       'A8J3230', 'A8J3330', 'A8J3430', 'A8K3130', 'A8K3230', 'A8K3330',
       'A8K3430', 'A95W330', 'A95W430', 'A9E8130', 'A9

In [377]:
zmb51_filtrado = zmb51_filtro.loc[zmb51_filtro.index.isin(fup_Material)]

In [378]:
zmb51_filtrado.columns

DatetimeIndex(['2021-01-01', '2021-01-02', '2021-01-03', '2021-01-04',
               '2021-01-05', '2021-01-06', '2021-01-07', '2021-01-08',
               '2021-01-09', '2021-01-10', '2021-01-11', '2021-01-12',
               '2022-01-01', '2022-01-02', '2022-01-03', '2022-01-04',
               '2022-01-05', '2022-01-06', '2022-01-07', '2022-01-08',
               '2022-01-09', '2022-01-10', '2022-01-11', '2022-01-12',
               '2023-01-01', '2023-01-02', '2023-01-03', '2023-01-04',
               '2023-01-05', '2023-01-06', '2023-01-07', '2023-01-08',
               '2023-01-09', '2023-01-10', '2023-01-11', '2023-01-12',
               '2024-01-01', '2024-01-02', '2024-01-03', '2024-01-04',
               '2024-01-05', '2024-01-06', '2024-01-07', '2024-01-08',
               '2024-01-09', '2024-01-10', '2024-01-11', '2024-01-12',
               '2025-01-01', '2025-01-02', '2025-01-03', '2025-01-04'],
              dtype='datetime64[ns]', name='Data de lançamento', freq=None)

In [379]:
zmb51_filtrado = zmb51_filtrado[['2024-01-11', '2024-01-12','2025-01-01', '2025-01-02', '2025-01-03', '2025-01-04']]
zmb51_filtrado

Data de lançamento  2024-01-11  2024-01-12  2025-01-01  2025-01-02  \
Material                                                             
14YK                       1.0        42.0         0.0         6.0   
A0TH500                   33.0        15.0        25.0        22.0   
A0X5134                   94.0        97.0        74.0       147.0   
A0X5135                    5.0        12.0         9.0         9.0   
A0X5234                   44.0        30.0        25.0        52.0   
...                        ...         ...         ...         ...   
AE2F430                   35.0         2.0         0.0        17.0   
AE2G700                    1.0         0.0         0.0         7.0   
AE2G800                    1.0         0.0         0.0         6.0   
AE2G900                    1.0         0.0         0.0         6.0   
AE7XWY1                    7.0        10.0         0.0        16.0   

Data de lançamento  2025-01-03  2025-01-04  
Material                                    
14YK                      18.0         0.0  
A0TH500                   30.0         0.0  
A0X5134                   80.0         2.0  
A0X5135                   26.0         0.0  
A0X5234                   16.0         0.0  
...                        ...         ...  
AE2F430                   53.0         0.0  
AE2G700                    1.0         0.0  
AE2G800                    2.0         0.0  
AE2G900                    2.0         0.0  
AE7XWY1                    8.0         0.0  

[120 rows x 6 columns]

In [380]:
zmb51_filtrado['Média'] = zmb51_filtrado.mean(axis=1)
zmb51_filtrado

Data de lançamento  2024-01-11 00:00:00  2024-01-12 00:00:00  \
Material                                                       
14YK                                1.0                 42.0   
A0TH500                            33.0                 15.0   
A0X5134                            94.0                 97.0   
A0X5135                             5.0                 12.0   
A0X5234                            44.0                 30.0   
...                                 ...                  ...   
AE2F430                            35.0                  2.0   
AE2G700                             1.0                  0.0   
AE2G800                             1.0                  0.0   
AE2G900                             1.0                  0.0   
AE7XWY1                             7.0                 10.0   

Data de lançamento  2025-01-01 00:00:00  2025-01-02 00:00:00  \
Material                                                       
14YK                                0.0                  6.0   
A0TH500                            25.0                 22.0   
A0X5134                            74.0                147.0   
A0X5135                             9.0                  9.0   
A0X5234                            25.0                 52.0   
...                                 ...                  ...   
AE2F430                             0.0                 17.0   
AE2G700                             0.0                  7.0   
AE2G800                             0.0                  6.0   
AE2G900                             0.0                  6.0   
AE7XWY1                             0.0                 16.0   

Data de lançamento  2025-01-03 00:00:00  2025-01-04 00:00:00      Média  
Material                                                                 
14YK                               18.0                  0.0  11.166667  
A0TH500                            30.0                  0.0  20.833333  
A0X5134                            80.0                  2.0  82.333333  
A0X5135                            26.0                  0.0  10.166667  
A0X5234                            16.0                  0.0  27.833333  
...                                 ...                  ...        ...  
AE2F430                            53.0                  0.0  17.833333  
AE2G700                             1.0                  0.0   1.500000  
AE2G800                             2.0                  0.0   1.500000  
AE2G900                             2.0                  0.0   1.500000  
AE7XWY1                             8.0                  0.0   6.833333  

[120 rows x 7 columns]

In [381]:
zstok.columns

Index(['Tipo de material', 'Material', 'Descrição', 'UMB', 'Cen.', 'Dep.',
       'Pos.dpst.', 'Utiliz.livre', 'Contr.qualid.', 'Transf.dep.',
       'Transfer.(centro)', 'Bloqueado', 'Estoque Total', 'Reservado',
       'Remessa->Cliente', 'SC', 'Status todos centros', 'SV',
       'Status vendas'],
      dtype='object')

In [382]:
zstok_filtro = pd.pivot_table(zstok,
    values='Estoque Total',
    index='Material',
    columns='Cen.',
    aggfunc='sum',
    fill_value=0
)
zstok_filtro

Cen.        CE01  CE03  CE04  CE05  CE07  CE08  CE09  CE10  CE11  CE13
Material                                                              
47           0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  85.0   0.0
200          0.0   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0   0.0
201          0.0   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0   0.0
202          0.0   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0   0.0
658          0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   4.0
...          ...   ...   ...   ...   ...   ...   ...   ...   ...   ...
ZA10438370   0.0   0.0   0.0   0.0   2.0   0.0   0.0   0.0   1.0   0.0
ZA1043837X   0.0   0.0   0.0   0.0   2.0   0.0   0.0   0.0   0.0   0.0
ZA104501A0   0.0   0.0   0.0   0.0   1.0   0.0   0.0   0.0   3.0   0.0
ZA10458100   0.0   0.0   0.0   0.0   1.0   0.0   0.0   0.0   1.0   0.0
ZA10461900   0.0   0.0   0.0   0.0   5.0   0.0   0.0   0.0   0.0   0.0

[7723 rows x 10 columns]

In [383]:
zmb51_filtrado = zmb51_filtrado.reset_index() 
fup_filtro = fup_filtro.reset_index() 
novo_df = zmb51_filtrado.merge(fup_filtro, on='Material', how='left')
novo_df

Material  2024-01-11 00:00:00  2024-01-12 00:00:00  2025-01-01 00:00:00  \
0       14YK                  1.0                 42.0                  0.0   
1    A0TH500                 33.0                 15.0                 25.0   
2    A0X5134                 94.0                 97.0                 74.0   
3    A0X5135                  5.0                 12.0                  9.0   
4    A0X5234                 44.0                 30.0                 25.0   
..       ...                  ...                  ...                  ...   
115  AE2F430                 35.0                  2.0                  0.0   
116  AE2G700                  1.0                  0.0                  0.0   
117  AE2G800                  1.0                  0.0                  0.0   
118  AE2G900                  1.0                  0.0                  0.0   
119  AE7XWY1                  7.0                 10.0                  0.0   

     2025-01-02 00:00:00  2025-01-03 00:00:00  2025-01-04 00:00:00      Média  \
0                    6.0                 18.0                  0.0  11.166667   
1                   22.0                 30.0                  0.0  20.833333   
2                  147.0                 80.0                  2.0  82.333333   
3                    9.0                 26.0                  0.0  10.166667   
4                   52.0                 16.0                  0.0  27.833333   
..                   ...                  ...                  ...        ...   
115                 17.0                 53.0                  0.0  17.833333   
116                  7.0                  1.0                  0.0   1.500000   
117                  6.0                  2.0                  0.0   1.500000   
118                  6.0                  2.0                  0.0   1.500000   
119                 16.0                  8.0                  0.0   6.833333   

     2025-04-11 00:00:00  
0                     23  
1                     10  
2                     95  
3                     11  
4                     31  
..                   ...  
115                   29  
116                    4  
117                    2  
118                    5  
119                   18  

[120 rows x 9 columns]

In [384]:
novo_df = novo_df.rename(columns={pd.Timestamp('2025-04-11'): 'fup'})
novo_df

Material  2024-01-11 00:00:00  2024-01-12 00:00:00  2025-01-01 00:00:00  \
0       14YK                  1.0                 42.0                  0.0   
1    A0TH500                 33.0                 15.0                 25.0   
2    A0X5134                 94.0                 97.0                 74.0   
3    A0X5135                  5.0                 12.0                  9.0   
4    A0X5234                 44.0                 30.0                 25.0   
..       ...                  ...                  ...                  ...   
115  AE2F430                 35.0                  2.0                  0.0   
116  AE2G700                  1.0                  0.0                  0.0   
117  AE2G800                  1.0                  0.0                  0.0   
118  AE2G900                  1.0                  0.0                  0.0   
119  AE7XWY1                  7.0                 10.0                  0.0   

     2025-01-02 00:00:00  2025-01-03 00:00:00  2025-01-04 00:00:00      Média  \
0                    6.0                 18.0                  0.0  11.166667   
1                   22.0                 30.0                  0.0  20.833333   
2                  147.0                 80.0                  2.0  82.333333   
3                    9.0                 26.0                  0.0  10.166667   
4                   52.0                 16.0                  0.0  27.833333   
..                   ...                  ...                  ...        ...   
115                 17.0                 53.0                  0.0  17.833333   
116                  7.0                  1.0                  0.0   1.500000   
117                  6.0                  2.0                  0.0   1.500000   
118                  6.0                  2.0                  0.0   1.500000   
119                 16.0                  8.0                  0.0   6.833333   

     fup  
0     23  
1     10  
2     95  
3     11  
4     31  
..   ...  
115   29  
116    4  
117    2  
118    5  
119   18  

[120 rows x 9 columns]

In [385]:
zstok_filtro = zstok_filtro.reset_index() 
zstok_filtro

Cen.    Material  CE01  CE03  CE04  CE05  CE07  CE08  CE09  CE10  CE11  CE13
0             47   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  85.0   0.0
1            200   0.0   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0   0.0
2            201   0.0   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0   0.0
3            202   0.0   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0   0.0
4            658   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   4.0
...          ...   ...   ...   ...   ...   ...   ...   ...   ...   ...   ...
7718  ZA10438370   0.0   0.0   0.0   0.0   2.0   0.0   0.0   0.0   1.0   0.0
7719  ZA1043837X   0.0   0.0   0.0   0.0   2.0   0.0   0.0   0.0   0.0   0.0
7720  ZA104501A0   0.0   0.0   0.0   0.0   1.0   0.0   0.0   0.0   3.0   0.0
7721  ZA10458100   0.0   0.0   0.0   0.0   1.0   0.0   0.0   0.0   1.0   0.0
7722  ZA10461900   0.0   0.0   0.0   0.0   5.0   0.0   0.0   0.0   0.0   0.0

[7723 rows x 11 columns]

In [386]:
zstok_filtro.columns

Index(['Material', 'CE01', 'CE03', 'CE04', 'CE05', 'CE07', 'CE08', 'CE09',
       'CE10', 'CE11', 'CE13'],
      dtype='object', name='Cen.')

In [387]:
zstok_filtro = zstok_filtro[['Material','CE03', 'CE04', 'CE05', 'CE08', 'CE09',
       'CE10', 'CE11', 'CE13']]
zstok_filtro

Cen.    Material  CE03  CE04  CE05  CE08  CE09  CE10  CE11  CE13
0             47   0.0   0.0   0.0   0.0   0.0   0.0  85.0   0.0
1            200   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0
2            201   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0
3            202   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0
4            658   0.0   0.0   0.0   0.0   0.0   0.0   0.0   4.0
...          ...   ...   ...   ...   ...   ...   ...   ...   ...
7718  ZA10438370   0.0   0.0   0.0   0.0   0.0   0.0   1.0   0.0
7719  ZA1043837X   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0
7720  ZA104501A0   0.0   0.0   0.0   0.0   0.0   0.0   3.0   0.0
7721  ZA10458100   0.0   0.0   0.0   0.0   0.0   0.0   1.0   0.0
7722  ZA10461900   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0

[7723 rows x 9 columns]

In [388]:
zstok_filtro = zstok_filtro.set_index('Material')
zstok_filtro['Total'] = zstok_filtro.sum(axis=1)
zstok_filtro = zstok_filtro.reset_index() 
zstok_filtro

Cen.    Material  CE03  CE04  CE05  CE08  CE09  CE10  CE11  CE13  Total
0             47   0.0   0.0   0.0   0.0   0.0   0.0  85.0   0.0   85.0
1            200   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0    7.0
2            201   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0    7.0
3            202   0.0   0.0   7.0   0.0   0.0   0.0   0.0   0.0    7.0
4            658   0.0   0.0   0.0   0.0   0.0   0.0   0.0   4.0    4.0
...          ...   ...   ...   ...   ...   ...   ...   ...   ...    ...
7718  ZA10438370   0.0   0.0   0.0   0.0   0.0   0.0   1.0   0.0    1.0
7719  ZA1043837X   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0    0.0
7720  ZA104501A0   0.0   0.0   0.0   0.0   0.0   0.0   3.0   0.0    3.0
7721  ZA10458100   0.0   0.0   0.0   0.0   0.0   0.0   1.0   0.0    1.0
7722  ZA10461900   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0    0.0

[7723 rows x 10 columns]

In [389]:
zstok_filtro2 = zstok_filtro[['Material','Total']]
zstok_filtro = zstok_filtro.rename(columns={'Total': 'Estoque'})


In [390]:
novo_df = novo_df.merge(zstok_filtro2, on='Material', how='left')
novo_df

Material  2024-01-11 00:00:00  2024-01-12 00:00:00  2025-01-01 00:00:00  \
0       14YK                  1.0                 42.0                  0.0   
1    A0TH500                 33.0                 15.0                 25.0   
2    A0X5134                 94.0                 97.0                 74.0   
3    A0X5135                  5.0                 12.0                  9.0   
4    A0X5234                 44.0                 30.0                 25.0   
..       ...                  ...                  ...                  ...   
115  AE2F430                 35.0                  2.0                  0.0   
116  AE2G700                  1.0                  0.0                  0.0   
117  AE2G800                  1.0                  0.0                  0.0   
118  AE2G900                  1.0                  0.0                  0.0   
119  AE7XWY1                  7.0                 10.0                  0.0   

     2025-01-02 00:00:00  2025-01-03 00:00:00  2025-01-04 00:00:00      Média  \
0                    6.0                 18.0                  0.0  11.166667   
1                   22.0                 30.0                  0.0  20.833333   
2                  147.0                 80.0                  2.0  82.333333   
3                    9.0                 26.0                  0.0  10.166667   
4                   52.0                 16.0                  0.0  27.833333   
..                   ...                  ...                  ...        ...   
115                 17.0                 53.0                  0.0  17.833333   
116                  7.0                  1.0                  0.0   1.500000   
117                  6.0                  2.0                  0.0   1.500000   
118                  6.0                  2.0                  0.0   1.500000   
119                 16.0                  8.0                  0.0   6.833333   

     fup  Total  
0     23   11.0  
1     10   38.0  
2     95  523.0  
3     11   14.0  
4     31  288.0  
..   ...    ...  
115   29   39.0  
116    4   10.0  
117    2   10.0  
118    5   10.0  
119   18   24.0  

[120 rows x 10 columns]

In [391]:
novo_df.columns

Index([         'Material', 2024-01-11 00:00:00, 2024-01-12 00:00:00,
       2025-01-01 00:00:00, 2025-01-02 00:00:00, 2025-01-03 00:00:00,
       2025-01-04 00:00:00,             'Média',               'fup',
                   'Total'],
      dtype='object')

In [392]:
novo_df = novo_df[['Material','Média','fup','Total']]

In [393]:
novo_df = novo_df.rename(columns={'Média': 'Media'})

In [394]:
novo_df

Material      Media  fup  Total
0       14YK  11.166667   23   11.0
1    A0TH500  20.833333   10   38.0
2    A0X5134  82.333333   95  523.0
3    A0X5135  10.166667   11   14.0
4    A0X5234  27.833333   31  288.0
..       ...        ...  ...    ...
115  AE2F430  17.833333   29   39.0
116  AE2G700   1.500000    4   10.0
117  AE2G800   1.500000    2   10.0
118  AE2G900   1.500000    5   10.0
119  AE7XWY1   6.833333   18   24.0

[120 rows x 4 columns]

In [395]:
json_str = novo_df.to_json(orient='records', indent=4)
print(json_str)

[
    {
        "Material":"14YK",
        "Media":11.1666666667,
        "fup":23,
        "Total":11.0
    },
    {
        "Material":"A0TH500",
        "Media":20.8333333333,
        "fup":10,
        "Total":38.0
    },
    {
        "Material":"A0X5134",
        "Media":82.3333333333,
        "fup":95,
        "Total":523.0
    },
    {
        "Material":"A0X5135",
        "Media":10.1666666667,
        "fup":11,
        "Total":14.0
    },
    {
        "Material":"A0X5234",
        "Media":27.8333333333,
        "fup":31,
        "Total":288.0
    },
    {
        "Material":"A0X5235",
        "Media":5.0,
        "fup":10,
        "Total":2.0
    },
    {
        "Material":"A0X5334",
        "Media":35.8333333333,
        "fup":39,
        "Total":242.0
    },
    {
        "Material":"A0X5335",
        "Media":5.1666666667,
        "fup":7,
        "Total":5.0
    },
    {
        "Material":"A0X5434",
        "Media":35.8333333333,
        "fup":41,
        "Total":288.0
 

In [396]:
#novo_df.to_json('novo_df.json', orient='records', indent=4)

In [397]:
zstok2 =zstok[['Material','Descrição']]

In [398]:
#novo_df = novo_df.merge(zstok2, on='Material', how='left')


In [399]:
novo_df["Estoque_proj"] = novo_df["Total"]

# Calcular a cobertura em meses
novo_df["Meses_cobertura"] = novo_df["Estoque_proj"] / novo_df["Media"]

# Verificar se haverá falta (cobertura menor que 1 mês)
novo_df["Faltara_esse_mes"] = novo_df["Meses_cobertura"] < 1

# Separar os materiais com risco de falta
df_falta = novo_df[novo_df["Faltara_esse_mes"] == True]

# Separar os materiais que não faltarão
df_ok = novo_df[novo_df["Faltara_esse_mes"] == False]
novo_df

Material      Media  fup  Total  Estoque_proj  Meses_cobertura  \
0       14YK  11.166667   23   11.0          11.0         0.985075   
1    A0TH500  20.833333   10   38.0          38.0         1.824000   
2    A0X5134  82.333333   95  523.0         523.0         6.352227   
3    A0X5135  10.166667   11   14.0          14.0         1.377049   
4    A0X5234  27.833333   31  288.0         288.0        10.347305   
..       ...        ...  ...    ...           ...              ...   
115  AE2F430  17.833333   29   39.0          39.0         2.186916   
116  AE2G700   1.500000    4   10.0          10.0         6.666667   
117  AE2G800   1.500000    2   10.0          10.0         6.666667   
118  AE2G900   1.500000    5   10.0          10.0         6.666667   
119  AE7XWY1   6.833333   18   24.0          24.0         3.512195   

     Faltara_esse_mes  
0                True  
1               False  
2               False  
3               False  
4               False  
..                ...  
115             False  
116             False  
117             False  
118             False  
119             False  

[120 rows x 7 columns]

In [400]:
df_ok

Material      Media  fup  Total  Estoque_proj  Meses_cobertura  \
1    A0TH500  20.833333   10   38.0          38.0         1.824000   
2    A0X5134  82.333333   95  523.0         523.0         6.352227   
3    A0X5135  10.166667   11   14.0          14.0         1.377049   
4    A0X5234  27.833333   31  288.0         288.0        10.347305   
6    A0X5334  35.833333   39  242.0         242.0         6.753488   
..       ...        ...  ...    ...           ...              ...   
115  AE2F430  17.833333   29   39.0          39.0         2.186916   
116  AE2G700   1.500000    4   10.0          10.0         6.666667   
117  AE2G800   1.500000    2   10.0          10.0         6.666667   
118  AE2G900   1.500000    5   10.0          10.0         6.666667   
119  AE7XWY1   6.833333   18   24.0          24.0         3.512195   

     Faltara_esse_mes  
1               False  
2               False  
3               False  
4               False  
6               False  
..                ...  
115             False  
116             False  
117             False  
118             False  
119             False  

[113 rows x 7 columns]

In [401]:
df_falta

Material      Media  fup  Total  Estoque_proj  Meses_cobertura  \
0       14YK  11.166667   23   11.0          11.0         0.985075   
5    A0X5235   5.000000   10    2.0           2.0         0.400000   
7    A0X5335   5.166667    7    5.0           5.0         0.967742   
45   A63W01F   7.000000    9    2.0           2.0         0.285714   
46   A63X03W   2.666667    6    0.0           0.0         0.000000   
75   A9E8130  24.000000   40   16.0          16.0         0.666667   
100  ACTD030   3.500000   18    1.0           1.0         0.285714   

     Faltara_esse_mes  
0                True  
5                True  
7                True  
45               True  
46               True  
75               True  
100              True

In [402]:
novo_df["Estoque_proj"] = novo_df["Total"]

# Calcular a cobertura em meses
novo_df["Meses_cobertura"] = novo_df["Estoque_proj"] / novo_df["Media"]

# Verificar se haverá falta com cobertura menor que 2 meses
novo_df["Faltara_ate_2_meses"] = novo_df["Meses_cobertura"] < 2

# Separar os materiais com risco de falta (cobertura menor que 2 meses)
df_falta_2m = novo_df[novo_df["Faltara_ate_2_meses"] == True]

# Separar os materiais com cobertura suficiente (2 meses ou mais)
df_ok_2m = novo_df[novo_df["Faltara_ate_2_meses"] == False]

In [403]:
df_falta_2m

Material        Media   fup   Total  Estoque_proj  Meses_cobertura  \
0       14YK    11.166667    23    11.0          11.0         0.985075   
1    A0TH500    20.833333    10    38.0          38.0         1.824000   
3    A0X5135    10.166667    11    14.0          14.0         1.377049   
5    A0X5235     5.000000    10     2.0           2.0         0.400000   
7    A0X5335     5.166667     7     5.0           5.0         0.967742   
9    A0X5435     4.833333    17     8.0           8.0         1.655172   
14   A2XN0ED     5.333333    25     9.0           9.0         1.687500   
40   A5E7700    22.000000    19    40.0          40.0         1.818182   
41   A5E7800    22.166667    25    39.0          39.0         1.759398   
44   A5WJ0Y0    80.166667   106   155.0         155.0         1.933472   
45   A63W01F     7.000000     9     2.0           2.0         0.285714   
46   A63X03W     2.666667     6     0.0           0.0         0.000000   
55   A85Y03D     6.666667    10     9.0           9.0         1.350000   
61   A8DA130  1237.000000  3630  2440.0        2440.0         1.972514   
75   A9E8130    24.000000    40    16.0          16.0         0.666667   
77   A9E8330     8.833333    32    13.0          13.0         1.471698   
90   AAV708D     5.500000     7     9.0           9.0         1.636364   
100  ACTD030     3.500000    18     1.0           1.0         0.285714   
101  ACVH130    83.500000   120   151.0         151.0         1.808383   

     Faltara_esse_mes  Faltara_ate_2_meses  
0                True                 True  
1               False                 True  
3               False                 True  
5                True                 True  
7                True                 True  
9               False                 True  
14              False                 True  
40              False                 True  
41              False                 True  
44              False                 True  
45               True                 True  
46               True                 True  
55              False                 True  
61              False                 True  
75               True                 True  
77              False                 True  
90              False                 True  
100              True                 True  
101             False                 True

In [404]:
df_ok_2m

Material      Media  fup  Total  Estoque_proj  Meses_cobertura  \
2    A0X5134  82.333333   95  523.0         523.0         6.352227   
4    A0X5234  27.833333   31  288.0         288.0        10.347305   
6    A0X5334  35.833333   39  242.0         242.0         6.753488   
8    A0X5434  35.833333   41  288.0         288.0         8.037209   
10   A2X203D   1.666667    3    6.0           6.0         3.600000   
..       ...        ...  ...    ...           ...              ...   
115  AE2F430  17.833333   29   39.0          39.0         2.186916   
116  AE2G700   1.500000    4   10.0          10.0         6.666667   
117  AE2G800   1.500000    2   10.0          10.0         6.666667   
118  AE2G900   1.500000    5   10.0          10.0         6.666667   
119  AE7XWY1   6.833333   18   24.0          24.0         3.512195   

     Faltara_esse_mes  Faltara_ate_2_meses  
2               False                False  
4               False                False  
6               False                False  
8               False                False  
10              False                False  
..                ...                  ...  
115             False                False  
116             False                False  
117             False                False  
118             False                False  
119             False                False  

[101 rows x 8 columns]

In [405]:
# Exportar os materiais com cobertura menor que 2 meses para JSON
df_falta_2m.to_json("materiais_com_falta_ate_2_meses.json", orient="records", indent=4)

#  Materiais com Falta Imediata (Cobertura < 1 mês)



## 1. Material: **14YK**
- **Descrição**: Grampo para unidade de acabamento FS-504, utilizado em impressoras Bizhub 600.
- **Estoque atual**: 11 unidades  
- **Média de consumo mensal**: 11,17 unidades  
- **Cobertura estimada**: 0,99 meses

🔎 **Risco**: Estoque cobre praticamente o mês atual, mas qualquer variação pode causar ruptura.

---

## 2. Material: **A0X5235**
- **Descrição**: Cartucho de toner amarelo TNP51Y, para impressora Bizhub C3110.
- **Estoque atual**: 2 unidades  
- **Média de consumo mensal**: 5 unidades  
- **Cobertura estimada**: 0,40 meses

🔎 **Risco**: Estoque suficiente para cerca de 12 dias. Risco alto de parada na impressão colorida.

---

## 3. Material: **A0X5335**
- **Descrição**: Cartucho de toner magenta TNP51M, também para Bizhub C3110.
- **Estoque atual**: 5 unidades  
- **Média de consumo mensal**: 5,17 unidades  
- **Cobertura estimada**: 0,97 meses

🔎 **Risco**: Estoque quase suficiente, mas ainda dentro da faixa crítica. Monitoramento diário é recomendado.

---

## 4. Material: **A63W01F**
- **Descrição**: Cartucho de toner preto TNP35, utilizado na impressora Bizhub 4000P.
- **Estoque atual**: 2 unidades  
- **Média de consumo mensal**: 7 unidades  
- **Cobertura estimada**: 0,29 meses

🔎 **Risco**: Estoque cobre menos de 1 semana. Alta criticidade.

---

## 5. Material: **A63X03W**
- **Descrição**: Unidade de imagem IUP17, utilizada nas impressoras Bizhub 4000P / 4700.
- **Estoque atual**: 0 unidades  
- **Média de consumo mensal**: 2,67 unidades  
- **Cobertura estimada**: 0,00 meses

🔎 **Risco**: Estoque esgotado. Operações dependentes deste item já estão paradas ou comprometidas.





# Materiais com Cobertura entre 1 e 2 Meses


## 1. Material: **A0TH500**
- **Descrição**: Revelador em pó DV-011 para impressoras Bizhub PRO 1200 / 1051
- **Estoque atual**: 38 unidades
- **Média de consumo mensal**: 20,83 unidades
- **Cobertura estimada**: 1,82 meses

🔍 Comentário: A margem é estreita. Ideal realizar reforço no próximo ciclo de compras.

---

## 2. Material: **A0X5135**
- **Descrição**: Cartucho de toner preto TNP51K para Bizhub C3110
- **Estoque atual**: 14 unidades
- **Média de consumo mensal**: 10,17 unidades
- **Cobertura estimada**: 1,38 meses

🔍 Comentário: Cobertura apertada. Recomenda-se monitoramento semanal.

---

## 3. Material: **A0X5435**
- **Descrição**: Cartucho de toner ciano TNP51C para Bizhub C3110
- **Estoque atual**: 8 unidades
- **Média de consumo mensal**: 4,83 unidades
- **Cobertura estimada**: 1,65 meses

🔍 Comentário: Importante manter alinhamento com os demais toners coloridos (magenta e amarelo).

---

## 4. Material: **A2XN0ED**
- **Descrição**: Unidade reveladora magenta DV512M
- **Estoque atual**: 9 unidades
- **Média de consumo mensal**: 5,33 unidades
- **Cobertura estimada**: 1,69 meses

🔍 Comentário: Essencial para impressão colorida. Necessita de reposição em breve.

---

## 5. Material: **A5E7700**
- **Descrição**: Revelador amarelo DV616Y para Bizhub Press C1100 / C6100
- **Estoque atual**: 40 unidades
- **Média de consumo mensal**: 22 unidades
- **Cobertura estimada**: 1,82 meses

🔍 Comentário: Produto com giro alto. Mesmo com bom volume, o tempo de cobertura é curto.

---

## 6. Material: **A5E7800**
- **Descrição**: Revelador magenta DV616M para Bizhub C1100 / C6100
- **Estoque atual**: 39 unidades
- **Média de consumo mensal**: 22,17 unidades
- **Cobertura estimada**: 1,76 meses

🔍 Comentário: Situação semelhante ao item anterior. Importante garantir reposição.

---

## 7. Material: **A5WJ0Y0**
- **Descrição**: Cilindro DU106 para Bizhub Pro C1060L / C2060
- **Estoque atual**: 155 unidades
- **Média de consumo mensal**: 80,17 unidades
- **Cobertura estimada**: 1,93 meses

🔍 Comentário: Margem segura por enquanto. Acompanhar desgaste em campo.

---

## 8. Material: **A85Y03D**
- **Descrição**: Unidade reveladora preta DV214K para Bizhub C287 / C266
- **Estoque atual**: 9 unidades
- **Média de consumo mensal**: 6,67 unidades
- **Cobertura estimada**: 1,35 meses

🔍 Comentário: Estoque apertado. Repor antes da virada do mês.

---

## 9. Material: **A8DA130**
- **Descrição**: Toner preto TN324K para Bizhub C368 / C308 / C258
- **Estoque atual**: 2440 unidades
- **Média de consumo mensal**: 1237 unidades
- **Cobertura estimada**: 1,97 meses

🔍 Comentário: Apesar do volume elevado, o consumo também é alto. Ficar atento.

---

## 10. Material: **A9E8330**
- **Descrição**: Cartucho de toner magenta TN514M
- **Estoque atual**: 13 unidades
- **Média de consumo mensal**: 8,83 unidades
- **Cobertura estimada**: 1,47 meses

🔍 Comentário: Toner colorido com risco iminente caso o bloqueio persista.

---

## 11. Material: **AAV708D**
- **Descrição**: Revelador em pó DV315Y
- **Estoque atual**: 9 unidades
- **Média de consumo mensal**: 5,5 unidades
- **Cobertura estimada**: 1,64 meses

🔍 Comentário: Equipamento pode parar caso não haja reforço no estoque no mês seguinte.

---

## 12. Material: **ACVH130**
- **Descrição**: Toner preto TN227K para Bizhub C227i / C287i
- **Estoque atual**: 151 unidades
- **Média de consumo mensal**: 83,5 unidades
- **Cobertura estimada**: 1,81 meses

🔍 Comentário: Alta rotação. Estoque será consumido rapidamente.

---
